### PR TITLE
Add V12 polaris-migrator documentation to doc site

### PR DESCRIPTION
**Pull request by lgriffee**
### WHY are these changes introduced?

So far we've documented all of our migrations on the https://polaris.shopify.com/tools/polaris-migrator however we are missing documentation for v12. 

### WHAT is this pull request doing?

Adds v12 migration documentation to the https://polaris.shopify.com/tools/polaris-migrator page.

### CODE REVIEW

File: polaris.shopify.com/content/tools/polaris-migrator.md [hunk 1]
```diff
@@ -37,6 +37,163 @@ npx @shopify/polaris-migrator <migration> <path>
...
+
+```sh
+npx @shopify/polaris-migrator v12-react-avatar-component <path>
+```
+
+<br />
+
+#### `v12-react-update-button-component`
+
+Consolidates `Button` boolean props to `variant` and `tone`. The `Button` component has been updated to replace deprecated `connectedDisclosure`, `outline`, `destructive`, `primary`, `primarySuccess`, `plain`, and `monochrome` props with a new `variant` prop that supports multiple variation options.
```
Comment (lgriffee) [inline polaris.shopify.com/content/tools/polaris-migrator.md]:
```suggestion
Consolidate `Button` boolean props to `variant` and `tone`. The `Button` component has been updated to replace deprecated `connectedDisclosure`, `outline`, `destructive`, `primary`, `primarySuccess`, `plain`, and `monochrome` props with a new `variant` prop that supports multiple variation options.
```

File: polaris.shopify.com/content/tools/polaris-migrator.md [hunk 2]
```diff
@@ -37,6 +37,163 @@ npx @shopify/polaris-migrator <migration> <path>
...
+
+```sh
+npx @shopify/polaris-migrator v12-react-avatar-component <path>
+```
+
+<br />
+
+#### `v12-react-update-button-component`
+
+Consolidate `Button` boolean props to `variant` and `tone`. The `Button` component has been updated to replace deprecated `connectedDisclosure`, `outline`, `destructive`, `primary`, `primarySuccess`, `plain`, and `monochrome` props with a new `variant` prop that supports multiple variation options.
```
Comment (laurkim) [inline polaris.shopify.com/content/tools/polaris-migrator.md:71]:
```suggestion
Consolidate `Button` boolean props. The `Button` component has been updated to replace deprecated `connectedDisclosure`, `outline`, `destructive`, `primary`, `primarySuccess`, `plain`, and `monochrome` props with a new `variant` prop that supports multiple variation options.
```

**Review by laurkim — APPROVED** [review]
💯

**Review by chloerice — APPROVED** [review]
This looks good to me! Let's add a cross link to the migration guide so folks know there's way more to the v12 migrations than the ones we grouped 😅

**Review by aaronccasanova — APPROVED** [review]

**Comment by lgriffee** [comment]
> This looks good to me! Let's add a cross link to the migration guide so folks know there's way more to the v12 migrations than the ones we grouped 😅

@chloerice Great idea! Added a cross link in 6245e0e!

**Pull request merged**